### PR TITLE
Add tests for acoustic module

### PR DESCRIPTION
**Pull request by jh66637**
I added some tests for the acoustic module. To be able to test different parameters, I moved a lot of parameters from the `application.h` file to the `input.json` file.

### CODE REVIEW

**Comment by nfehn** [comment]
Do we really need ABM1 to ABM4? In `exadg/tests`, you already added the `ab` and `am` tests. What would you expect from the additional tests here? 

One could argue that tests are always good. However, at the same time we need to make sure that the number of tests does not become excessive or unnecessarily high. In the end, it is in the interest of all of us that the tests run quickly, and that we do not have to change a lot regarding testing when making changes to the code. So I appreciate the addition of tests, but at the same time I argue for a minimal set of tests.

We have many many explicit RK time integrators in ExaDG, which could be used for the acoustics module as well. Similarly, the ABM time integrators could be used for CompNS. It is not practical to add tests for every time integrator in my opinion.

**Comment by nfehn** [comment]
How long does a single test run?

Background: Generally, we follow the goal that a single test should not run longer than 20-30 seconds. The idea behind is again that we want to ensure/allow fast development.

**Comment by jh66637** [comment]
Each test takes less than 10 seconds on my laptop.

The existing tests only check if the multistep constants are correct. By now, it is not checked if the use of these constants is correct. In this case, one could for example make a mistake during the prediction, correction and push back of solution vectors. I fixed a bug related to this quite recently. With the bug, the convergence orders were correct, but the time step size was limited roughly to half of the time step size one can use without the bug. This is why I would like to keep the tests.

File: applications/acoustic_conservation_equations/vibrating_membrane/tests/abm1.output
Comment (jh66637) [reply applications/acoustic_conservation_equations/vibrating_membrane/tests/abm1.output]: [2579276](https://github.com/exadg/exadg/pull/603/commits/25792763e151753c78bf1a74d2bcc9141cd1df67)

**Comment by nfehn** [comment]
10 seconds is absolutely fine. Nice!

I understand the wish/need to also test time integration.

But: When looking at `Acoustics::TimeIntAdamsBashforthMoulton` (https://github.com/exadg/exadg/blob/master/include/exadg/acoustic_conservation_equations/time_integration/time_int_abm.h), this class essentially contains the function `calculate_time_step_size()` (the other functions related to postprocessing and printing solver information will not be visible /checked during tests if `is_test = true`). But `calculate_time_step_size()` is in my understanding not related to the order of the time integration scheme. When looking at the base class `TimeIntAdamsBashforthMoultonBase` (https://github.com/exadg/exadg/blob/master/include/exadg/time_integration/time_int_abm_base.h), which part of that code depends explicitly on the order of the time integration scheme? In my opinion, the order seems to be completely hidden behind the time integration constants, which are tested elsewhere. In this regard, it is still not clear to me why 4 tests for orders 1-4 are necessary, in particular on the global application level. In my opinion, a single test for one particular order would cover the same code. Maybe you could explain the bug you described a bit more so that I know which aspect of the time integration is critical to check.

I will not argue that you have to remove these tests here because I appreciate a lot that you are writing tests. However, I am against increasing the number of tests at this speed if there is no clear reasoning that different tests aim to cover different aspects of the implementation.

Do you have an explanation why the velocity error at `dt=2.5e-4` is smaller for ABM2 than for ABM3? Or did you change also other parameters between these two tests?

**Review by nfehn — APPROVED** [review]

**Comment by jh66637** [comment]
I see what you mean. I removed all tests except the ones for order 3.

The larger velocity error for order 3 is something I observed before in some configurations of this test case. I think the problem is, that the test case is very easy to approximate for high-order FE methods. When looking at convergence rates I always had to increase the number of modes to be able to show the convergence rates on the triangulation that is used in the application. 

In any case: If you look closely, the overall error (pressure and velocity) still decreases as expected.

**Comment by nfehn** [comment]
I will merge once the tests passed.

**Comment by jh66637** [comment]
Can you automatically squash before merging or do I have to squash manually?

**Comment by jh66637** [comment]
OK, I squashed manually :)

**Pull request merged**
